### DISCUSSION TITLE
next/image with different aspect ratios per breakpoint

### DISCUSSION THREAD

**Discussion by jobe451** [vercel/next.js#34184] [Help]
So I have these beautiful hero-images at the top of each page fixed positioned. Part of it is hidden by menu and content. While scrolling up, the user "explores" all of it. In order that this all works nicely at different screen sizeses / mobile I need different aspect ratios.

So for mobile I use an aspect ratio of 4:3, on large screen it is rather 2/1. So far I prepared different explicit images for that pupose and displayed them with img srcset.

It would be nice toto automatize that and simply provide a large 4.3 image and have next/image do the magic. In order for that to happen, I would need to write somewhat a custom loader. The loader only returns an URL to a service that creates those images. So I guess I would need to write my own service that does the formatting and croping. 

Does something like that already exist? Any ideas how to approach that?

**Comment by kkggg55** [comment]
agreed, I hope next.js can push this in the next update. However for the meantime, you can use shadcn aspect ratio component which uses Radix to do it: https://ui.shadcn.com/docs/components/aspect-ratio
